plugin/helpers_bots.go

```diff
@@ -72,3 +105,50 @@ func (p *HelpersImpl) EnsureBot(bot *model.Bot) (retBotId string, retErr error)
 
 	return createdBot.UserId, nil
 }
+
+func (p *HelpersImpl) ShouldProcessMessage(post *model.Post, options ...ShouldProcessMessageOption) (bool, error) {
+	messageProcessOptions := &shouldProcessMessageOptions{}
+	for _, option := range options {
+		option(messageProcessOptions)
+	}
+
+	if !messageProcessOptions.AllowSystemMessages {
+		if post.IsSystemMessage() {
+			return false, nil
+		}
+	}
+
+	if len(messageProcessOptions.FilterChannelIDs) != 0 && !utils.StringInSlice(post.ChannelId, messageProcessOptions.FilterChannelIDs) {
+		channel, appErr := p.API.GetChannel(post.ChannelId)
+
+		if appErr != nil {
+			return false, errors.Wrap(appErr, "unable to get channel")
+		}
+
+		botIdBytes, kvGetErr := p.API.KVGet(BOT_USER_KEY)
+		if kvGetErr != nil || botIdBytes == nil {
+			return false, errors.Wrap(kvGetErr, "failed to get bot")
+		}
+
+		botId := string(botIdBytes)
+		if !model.IsBotDMChannel(channel, botId) {
+			return false, nil
+		}
+	}
+
+	if len(messageProcessOptions.FilterUserIDs) != 0 && !utils.StringInSlice(post.UserId, messageProcessOptions.FilterUserIDs) {
+		user, appErr := p.API.GetUser(post.UserId)
+
+		if appErr != nil {
+			return false, errors.Wrap(appErr, "unable to get user")
+		}
+
+		if !messageProcessOptions.AllowBots {
```

Comment:
I don't think `AllowBots` should only be checked when `FilterUserIDs` is non-empty. My thinking for the overall semantics of this method are:
```
// is_system_message && !allow_system_messages => return false
// is_bot && !allow_bots => return false
// filter_channel_ids not empty && channel not in filter_channel_ids => return false
// filter_user_ids not empty && user not in filter_user_ids => return false
// user is my own bot => return false
```